Dockerfile

```diff
@@ -37,8 +46,6 @@ ENV PYTHONUNBUFFERED=1 \
 EXPOSE 80
 
 # Install the application server.
-RUN pip install "gunicorn>=20.1.0,<20.2.0"
-
-COPY . .
+RUN pip install .[production]
```

Comment:
should this be `pdm install -G production`?

Reply:
No need for `pdm` here, since the `production` group is part of the package's `optional-dependencies`


... except I forgot to add that group. Fixing now...

Reply:
Shouldn't the docker image be built from the source at hand, rather than rely on whatever package was most recently pushed to pypi?

Reply:
It's still installing from the local source, by specifying the `.`

Reply:
ah, right!

But what's the reason for `pip` over `pdm`? Seems less consistent.

Reply:
No need for any dev dependencies. No need to install PDM. Ensures the production version is fully PEP compliant.